internal/ingress/annotations/auth/main.go

```diff
@@ -108,11 +109,21 @@ func (a auth) Parse(ing *extensions.Ingress) (interface{}, error) {
 		}
 	}
 
-	name := fmt.Sprintf("%v/%v", ing.Namespace, s)
-	secret, err := a.r.GetSecret(name)
+	sns, sname, err := cache.SplitMetaNamespaceKey(s)
 	if err != nil {
 		return nil, ing_errors.LocationDenied{
-			Reason: errors.Wrapf(err, "unexpected error reading secret %v", name),
+			Reason: errors.Wrap(err, "error reading secret name from annotation"),
+		}
+	}
+
+	if sns == "" {
+		sname = fmt.Sprintf("%v/%v", ing.Namespace, sname)
```

Comment:
I think this should be `sns = ing.Namespace`

and then after the `if` block we should do

```
name = fmt.Sprintf("%v/%v", sns, sname)
```

Reply:
done